hspec-discover/test/Test/Hspec/Discover/SortSpec.hs

```diff
@@ -0,0 +1,49 @@
+module Test.Hspec.Discover.SortSpec (main, spec) where
+
+import           Helper
+
+import           Data.List (sortBy)
+import           Data.Ord (comparing)
+
+import           Test.Hspec.Discover.Sort
+
+main :: IO ()
+main = hspec spec
+
+
+spec :: Spec
+spec = do
+  describe "naturalSortKey" $ do
+    context "compares" $ do
+      it "reflexively" $ do
+        "" <=> "" `shouldBe` EQ
+        "Hello2World" <=> "Hello2World" `shouldBe` EQ
+        "123Hello456" <=> "123Hello456" `shouldBe` EQ
+        "" <=> "" `shouldBe` EQ
```

Comment:
Provided you derive the instance, I don't think you will need this test.  But if it were that you really want to test this, then this would be a prime use case for QuickCheck, e.g.:

```haskell
it "is reflexive" $ do
  property $ \ x -> x <=> x `shouldBe` EQ
```